### PR TITLE
Community feedback: Remove explicit cancellation

### PR DESCRIPTION
**Pull request by parkera**
Updates from [community feedback](https://forums.swift.org/t/se-0298-async-await-sequences). Thank you!

### CODE REVIEW

**Comment by lattner** [comment]
+1, thanks Tony

**Comment by DougGregor** [comment]
Thanks, Tony. I'm going to hold off on merging this right now because we generally don't want the proposal to change substantially in the middle of a review.

**Comment by parkera** [comment]
Sounds good to me.

File: proposals/0298-asyncsequence.md [hunk 1]
```diff
@@ -100,7 +100,7 @@ The standard library will define the following protocols:
 public protocol AsyncSequence {
   associatedtype AsyncIterator: AsyncIteratorProtocol where AsyncIterator.Element == Element
   associatedtype Element
-  func makeAsyncIterator() -> AsyncIterator
+  __consuming func makeAsyncIterator() -> AsyncIterator
 }
 
 public protocol AsyncIteratorProtocol {
```
Comment (benrimmington) [inline proposals/0298-asyncsequence.md:106]: @parkera `AsyncIteratorProtocol` still contains `__consuming mutating func cancel()` on line 109.

Comment (parkera) [reply proposals/0298-asyncsequence.md:106]: Yah, thanks. I'm going to fix that with a few other things.

File: proposals/0298-asyncsequence.md [hunk 2]
```diff
@@ -51,14 +51,16 @@ extension URL {
...
   for try await line in myFile.lines() {
-    header = line
-    break
+    if line.count > 80 {
+      longLine = line
+      break
+    }
   }
 } catch {
   header = nil // file didn't exist
```
Comment (benrimmington) [inline proposals/0298-asyncsequence.md:66]:
```suggestion
  longLine = nil // file didn't exist
```

File: proposals/0298-asyncsequence.md [hunk 3]
```diff
@@ -85,7 +87,7 @@ extension URL {
   struct Lines : AsyncSequence { }
 
   func lines() -> Lines
-  func firstLine() throws async -> String?
+  func firstLongLine() throws async -> String?
   func collectLines() throws async -> [String]
```
Comment (benrimmington) [inline proposals/0298-asyncsequence.md:91]:
```suggestion
  func firstLongLine() async throws -> String?
  func collectLines() async throws -> [String]
```

File: proposals/0298-asyncsequence.md [hunk 4]
```diff
@@ -182,7 +178,7 @@ The compiler will emit the equivalent of the following code:
 
 ```swift
 var it = myFile.lines().makeAsyncIterator()
-while let value = try await it.next() {
+while let line = try await it.next() {
   // Do something with each line
 }
```
Comment (benrimmington) [inline proposals/0298-asyncsequence.md:183]: Use `do` statement?

```swift
do {
  var it = myFile.lines().makeAsyncIterator()
  while let line = try await it.next() {
    // Do something with each line
  }
}
```

**Pull request merged**